internal/auth/client.go

```
@@ -0,0 +1,485 @@
+		metadata.TokenEndpoint,
+		metadata.RevocationEndpoint,
+	} {
+		if _, err := normalizeRegistryURL(endpoint); err != nil {
+			return Credentials{}, fmt.Errorf("unsafe OAuth endpoint: %w", err)
+		}
+	}
+	if metadata.AuthorizationEndpoint == "" || metadata.TokenEndpoint == "" || metadata.RevocationEndpoint == "" {
+		return Credentials{}, errors.New("OAuth provider metadata is incomplete")
+	}
+
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)